gobblin-core/src/main/java/gobblin/writer/commands/MySqlBufferedInserter.java

```diff
@@ -0,0 +1,194 @@
+/*
+ * Copyright (C) 2014-2016 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package gobblin.writer.commands;
+
+import gobblin.configuration.ConfigurationKeys;
+import gobblin.configuration.State;
+import gobblin.converter.jdbc.JdbcEntryData;
+import gobblin.converter.jdbc.JdbcEntryDatum;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.util.List;
+import java.util.concurrent.Callable;
+import java.util.concurrent.TimeUnit;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.github.rholder.retry.Retryer;
+import com.github.rholder.retry.RetryerBuilder;
+import com.github.rholder.retry.StopStrategies;
+import com.github.rholder.retry.WaitStrategies;
+import com.google.common.base.Joiner;
+import com.google.common.collect.Lists;
+
+/**
+ * The implementation of JdbcBufferedInserter for MySQL.
+ * This purpose of buffered insert is mainly for performance reason and the implementation is based on the
+ * reference manual http://dev.mysql.com/doc/refman/5.0/en/insert-speed.html
+ */
+public class MySqlBufferedInserter implements JdbcBufferedInserter {
```

Comment:
The implementation is to use multi values in one insert statement as per MySQL recommendation. I've briefly tested out PreparedStatement.addBatch but it didn't show as performant as passing multi value within insert statement.